src/api/spec/models/project_spec.rb

```diff
@@ -287,4 +287,99 @@
       }
     end
   end
+
+  describe '.deleted?' do
+    it 'returns false if the project exists in the app' do
+      expect(Project.deleted?(project.name)).to be_falsey
+    end
+
+    it 'returns false if backend responds with nothing' do
+      allow_any_instance_of(ProjectFile).to receive(:to_s).with(deleted: 1).and_return(nil)
+      expect(Project.deleted?('never-existed-before')).to be_falsey
+    end
+
+    it 'returns false if revision list element of _history file is empty' do
+      allow_any_instance_of(ProjectFile).to receive(:to_s).with(deleted: 1).and_return("<revisionlist>\n</revisionlist>\n")
+      expect(Project.deleted?('never-existed-before')).to be_falsey
+    end
+
+    it 'returns true if _history element has elements' do
+      allow_any_instance_of(ProjectFile).to receive(:to_s).with(deleted: 1).and_return(
+        "<revisionlist>\n  <revision rev=\"1\" vrev=\"\">\n    <srcmd5>d41d8cd98f00b204e9800998ecf8427e</srcmd5>\n    " \
+        "<version></version>\n    <time>1498113679</time>\n    <user>Admin</user>\n    <comment>1</comment>\n  " \
+        "</revision>\n</revisionlist>\n"
+      )
+
+      expect(Project.deleted?('very-nice-project-name')).to be_truthy
+    end
+  end
+
+  describe '.restore' do
+    let(:admin_user) { create(:admin_user, login: 'Admin') }
+    let(:deleted_project) {
+      create(
+        :project_with_packages,
+        {
+          name:                'project_used_for_restoration',
+          title:               'restoration_project_title',
+          package_title:       'restoration_title',
+          package_description: 'restoration_desc',
+          package_name:        'restoration_package'
+        }
+      )
+    }
+
+    before do
+      login admin_user
+      deleted_project.destroy!
+    end
+
+    describe 'history element' do
+      it 'sets the user that restored the project' do
+        Project.restore(deleted_project.name, user: admin_user.login)
+
+        meta = Xmlhash.parse(ProjectFile.new(project_name: deleted_project.name, name: '_history').to_s(deleted: 1))
+        expect(meta['revision'].last['user']).to eq(admin_user.login)
+      end
+    end
+
+    describe 'project meta' do
```

Comment:
Also here, this describe could be part of the it. And if you use the meta method of the Project you can get rid of the let also